src/app/shared/services/principles.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+
+import { IPrinciple, Principle } from '../principle.model';
+
+import principlesData from '../../../assets/tables/principles-data.json';
+import { Observable, of } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class PrinciplesService {
+    principles: IPrinciple[];
+
+    private principlesByCategory: { [categoryName: string]: Principle[] } = {};
```

Comment:
nice job creating an anonymous type for this!

Reply:
Thanks, I'm actually proud of figuring out how to do it.  I really like having things hard typed, especially when the IDE warns you that you aren't returning the correct things.